os_net_config/impl_nmstate.py

```diff
@@ -1570,7 +1570,7 @@ def add_ib_child_interface(self, ib_child_interface):
         data[InfiniBand.CONFIG_SUBTREE] = config
         self.interface_data[ib_child_interface.name] = data
 
-    def apply(self, cleanup=False, activate=True):
+    def apply(self, cleanup=False, activate=True, config_rules_dns=True):
```

Comment:
We need to add a comment at L1583 that defines the config_rules_dns:

        :param config_rules_dns: A boolean that indicates if the rules should
            be applied. This makes sure that the rules are configured only if
            config_rules_dns is set to True.